edgelet/iotedge/src/main.rs

```diff
@@ -310,19 +340,117 @@ fn run() -> Result<(), Error> {
                 .value_of("tail")
-                .and_then(|a| a.parse::<LogTail>().ok())
-                .unwrap_or_default();
+                .map(str::parse)
+                .transpose()
+                .map_err(|err: edgelet_core::Error| {
+                    Error::from(err.context(ErrorKind::BadTailParameter))
+                })?
+                .expect("arg has a default value");
             let since = args
                 .value_of("since")
-                .and_then(|a| a.parse::<i32>().ok())
-                .unwrap_or_default();
+                .map(|s| parse_since(s))
+                .transpose()?
+                .expect("arg has a default value");
             let options = LogOptions::new()
                 .with_follow(follow)
                 .with_tail(tail)
                 .with_since(since);
             tokio_runtime.block_on(Logs::new(id, options, runtime()?).execute())
         }
+        ("support-bundle", Some(args)) => {
+            let location = args.value_of_os("output").expect("arg has a default value");
+            let since = args
+                .value_of("since")
+                .map(|s| parse_since(s))
+                .transpose()?
+                .expect("arg has a default value");
+            let options = LogOptions::new()
+                .with_follow(false)
+                .with_tail(LogTail::All)
+                .with_since(since);
+            let include_ms_only = args.is_present("include-edge-runtime-only");
+            tokio_runtime.block_on(
+                SupportBundle::new(options, location.to_owned(), include_ms_only, runtime()?)
+                    .execute(),
+            )
+        }
         ("version", _) => tokio_runtime.block_on(Version::new().execute()),
         (command, _) => tokio_runtime.block_on(Unknown::new(command.to_string()).execute()),
     }
 }
+
+fn parse_since(since: &str) -> Result<i32, Error> {
+    if let Ok(datetime) = DateTime::parse_from_rfc3339(since) {
+        let temp: Result<i32, _> = datetime.timestamp().try_into();
+        Ok(temp.context(ErrorKind::BadSinceParameter)?)
+    } else if let Ok(epoch) = since.parse() {
+        Ok(epoch)
+    } else if let Ok(duration) = parse_duration::parse(since) {
+        let nano: Result<i64, _> = duration.as_nanos().try_into();
+        let nano = nano.context(ErrorKind::BadSinceParameter)?;
+
+        let temp: Result<i32, _> = (Local::now() - Duration::nanoseconds(nano))
+            .timestamp()
+            .try_into();
+        Ok(temp.context(ErrorKind::BadSinceParameter)?)
+    } else {
+        Err(Error::from(ErrorKind::BadSinceParameter))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_rfc3339() {
+        assert_eq!(
+            parse_since("2019-09-27T16:00:00+00:00").unwrap(),
+            1_569_600_000
+        );
+    }
+
+    #[test]
+    fn parse_english() {
+        assert_near(
+            parse_since("1 hour").unwrap(),
```

Comment:
Does the customer need to put this duration in quotes on the command line or is clap smart enough to parse it?

Reply:
if there's spaces they will, but 1d or 2d6h works as well